syntax = "proto3";
package protos.playbooks;

import "google/protobuf/wrappers.proto";

message Github {
  message FetchRelatedCommits {
    google.protobuf.StringValue owner = 1;
    google.protobuf.StringValue repo = 2;
    google.protobuf.StringValue file_path = 3;
    google.protobuf.StringValue function_name = 4;
  }

  message FetchFile {
    google.protobuf.StringValue owner = 1;
    google.protobuf.StringValue repo = 2;
    google.protobuf.StringValue file_path = 3;
  }

  message UpdateFile {
    google.protobuf.StringValue owner = 1;
    google.protobuf.StringValue repo = 2;
    google.protobuf.StringValue file_path = 3;
    google.protobuf.StringValue committer_name = 4;
    google.protobuf.StringValue committer_email = 5;
    google.protobuf.StringValue content = 6;
    google.protobuf.StringValue message = 7;
    google.protobuf.StringValue sha = 8;
    google.protobuf.StringValue branch_name = 9;
  }

  enum TaskType {
    UNKNOWN = 0;
    FETCH_RELATED_COMMITS = 1;
    FETCH_FILE = 2;
    UPDATE_FILE = 3;
  }

  TaskType type = 1;
  oneof task {
    FetchRelatedCommits fetch_related_commits = 2;
    FetchFile fetch_file = 3;
    UpdateFile update_file = 4;
  }
}